bikespace_frontend/src/components/submission/location/Location.tsx

```diff
@@ -1,41 +1,68 @@
-import React from 'react';
+import React, {RefObject, useRef, useEffect} from 'react';
+import MapGL, {
+  Marker,
+  AttributionControl,
+  GeolocateControl,
+} from 'react-map-gl/maplibre';
+
 import {LocationLatLng} from '@/interfaces/Submission';
-import {MapContainer, TileLayer, Marker} from 'react-leaflet';
-import {LatLngTuple} from 'leaflet';
 
-import 'leaflet/dist/leaflet.css';
-import 'leaflet-defaulticon-compatibility';
-import 'leaflet-defaulticon-compatibility/dist/leaflet-defaulticon-compatibility.css';
+import 'maplibre-gl/dist/maplibre-gl.css';
 
 import styles from './location.module.scss';
 
 export interface LocationProps {
   location: LocationLatLng;
-  handler: React.ReactNode;
+  setLocation: React.Dispatch<React.SetStateAction<LocationLatLng>>;
 }
 
-function Location({location, handler}: LocationProps) {
-  const position = [location.latitude, location.longitude] as LatLngTuple;
+function Location({location, setLocation}: LocationProps) {
+  const geoControlRef =
+    useRef<maplibregl.GeolocateControl>() as RefObject<maplibregl.GeolocateControl>;
+
+  const {latitude, longitude} = location;
+
+  useEffect(() => {
+    if (!geoControlRef.current) return;
+
+    geoControlRef.current.trigger();
+  }, [geoControlRef.current]);
 
   return (
     <div className={styles.location}>
       <h2>Where was the problem?</h2>
       <h3>Pin the location</h3>
 
       <section className={styles.outerMapContainer} role="application">
-        <MapContainer
-          center={position}
-          zoom={18}
-          scrollWheelZoom={false}
-          style={{height: '100%'}}
+        <MapGL
+          initialViewState={{
+            latitude,
+            longitude,
+            zoom: 16,
```

Comment:
Might be best to keep zoom at 18, zoom 16 is a little too zoomed out for the scale at which someone would be clicking a bike parking location.